src/IAgreementFramework.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+import {AgreementParams, PositionParams} from "./AgreementStructs.sol";
+import {IArbitrable} from "./IArbitrable.sol";
+
+/// @notice Interface for agreemnts frameworks.
+/// @dev Implementations must manage the logic to manage individual agreements inside the same framework.
+interface IAgreementFramework is IArbitrable {
+
+    /* ====================================================================== //
+                                        EVENTS
+    // ====================================================================== */
+
+    event AgreementCreated(uint256 agreementId, bytes32 termsHash, uint256 criteria);
+    event AgreementJoined(uint256 agreementId, address party, uint256 balance);
+    event AgreementTermination(uint256 agreementId, address party);
```

Comment:
What about `AgreementFinalizationSigned`?

Reply:
@aahna-ashina we want to rename from `terminate` since it has a somewhat negative connotation (as in not amicable, like terminating someone's contract means firing them)